src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -984,9 +984,6 @@ public void visitClassDef(JCClassDecl tree) {
                         env.tree.hasTag(NEWCLASS)) {
                     c.flags_field |= NOOUTERTHIS;
                 }
-                if (env.tree.hasTag(NEWCLASS) && types.isPrimitiveClass(c.getSuperclass())) {
-                    c.flags_field |= PRIMITIVE_CLASS; // avoid further secondary errors.
-                }
                 attribClass(tree.pos(), c);
```

Comment:
This chunk purportedly introduced to serve the purpose of "avoiding further secondary errors" had outlived its purpose and is removed now. This change is not "essential" to the overall implementation and is a incidental cleanup